modules/dns_common/dns_common.rb

```diff
@@ -170,10 +170,10 @@ def aaaa_record_conflicts(fqdn, ip)
       end
     end
 
-    def cname_record_conflicts(fqdn, target)
-      current = resolver.getresources(fqdn, Resolv::DNS::Resource::IN::CNAME)
+    def cname_record_conflicts(fqdn, host_alias)
+      current = resolver.getresources(host_alias, Resolv::DNS::Resource::IN::CNAME)
```

Comment:
To me the DNS API is not DNS API but a Host to DNS translation API. That makes it inconsistent and hard to say which is 'correct' because it is a custom format.

The way I'd want it is that fqdn refers to what you query the server with and value what the server answers with. Since you query alias.example.com, that should be the value. In short: ```dig +short $type $fqdn``` should result in ```$value```. However, for PTRs this is already reversed. The way Foreman defines it this is that fqdn is the Host object and value is how it should point to the host object.

The current implementation already aligns with A/AAAA records. I'd say that PTR should be aligned with A/AAAA/CNAME rather than align CNAME with PTR. However, this is all breaking the API which we just can't do since this is all in released versions.

I'd propose to internally standardize the naming:
* ```fqdn``` is what is defined in the API and should only be in dns_api.rb
* ```value is what is defined in the API and should only be in dns_api.rb
* ```name``` is the DNS name you query for
* ```content``` is the result your DNS server should return

We should leave the current argument orders since this is the plugin API and breaking them is something we should avoid.